vortex-array/src/arrays/struct_/mod.rs

```diff
@@ -140,6 +140,25 @@ impl ArrayImpl for StructArray {
     fn _vtable(&self) -> VTableRef {
         VTableRef::new_ref(&StructEncoding)
     }
+
+    fn _with_children(&self, children: &[ArrayRef]) -> VortexResult<Self> {
+        let validity = if self.validity().is_array() {
```

Comment:
sometimes we `is_array` and sometimes we match...

Reply:
now we just check, the match seemed weird IMO